src/compiler/checker.ts

```diff
@@ -35345,13 +35357,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
         const restType = getTypeOfSymbol(restParameter);
         if (isTupleType(restType)) {
-            const associatedNames = ((restType as TypeReference).target as TupleType).labeledElementDeclarations;
+            const associatedNames = restType.target.labeledElementDeclarations;
             const index = pos - paramCount;
             const associatedName = associatedNames?.[index];
             const isRestTupleElement = !!associatedName?.dotDotDotToken;
 
-            if (associatedName) {
-                Debug.assert(isIdentifier(associatedName.name));
+            // TODO: figure out the non-identifier case here

```

Comment:
@MariaSolOs what kind of inlay hint interactivity is expected out of a parameter name?

Reply:
The expected behaviour is to be able to click on an argument of a function call and have that be a "go to definition" to the respective parameter in the function signature.